pyensembl/serializable.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+class Serializable(object):
+    """
+    Base class for all PyEnsembl objects which provides default
+    methods such as to_json, from_json, __reduce__, and from_dict
+    and relies only on (1) a user-defined to_dict method and (2) that the
+    keys of to_dict match the arguments to __init__.
+    """
+
+    def __str__(self):
+        fields_str = ", ".join(
+            "%s=%s" % (k, v) for (k, v) in sorted(self.to_dict().items()))
```

Comment:
I could see a use-case for not wanting to rely on sorting, but rather on some instrinsic ordering (e.g. ID first, name second, etc.). Any interest in using `OrderedDict`s instead?